src/lib-csharp/Logging/FileVelopackLogger.cs

```diff
@@ -0,0 +1,51 @@
+﻿#pragma warning disable CS1591 // Missing XML comment for publicly visible type or member
+using System;
+using System.Diagnostics;
+using System.IO;
+
+namespace Velopack.Logging
+{
+    public class FileVelopackLogger : IVelopackLogger, IDisposable
+    {
+        public uint ProcessId { get; }
+        private readonly object _lock = new();
+        private readonly StreamWriter _writer;
+        private readonly FileStream _fileStream;
+
+        public FileVelopackLogger(string filePath, uint processId)
+        {
+            ProcessId = processId;
+            _fileStream = new FileStream(filePath, FileMode.Append, FileAccess.Write, FileShare.ReadWrite);
+            _writer = new StreamWriter(_fileStream);
+        }
+
+        public void Log(VelopackLogLevel logLevel, string? message, Exception? exception)
+        {
+            try {
+                lock (_lock) {
+                    var logMessage = $"[lib-csharp:{ProcessId}] [{DateTime.Now.ToShortTimeString()}] [{logLevel}] {message}";
+                    if (exception != null) {
+                        logMessage += Environment.NewLine + exception;
+                    }
+
+                    _writer.WriteLine(logMessage);
+                }
```

Comment:
Wondering if there is a threading issue here. If a call to `Log` blocks on the `lock`, while a `Dispose` call is in progress, this then unblocks and tries to call `_writer.WriteLine` which will throw an `ObjectDisposedException`